internal/core/entities/game_room/spec.go

```diff
@@ -22,15 +22,34 @@
 
 package game_room
 
-import "time"
+import (
+	"github.com/Masterminds/semver"
+
+	"time"
+)
 
 type Spec struct {
-	Version                string        `validate:"min=1"`
-	TerminationGracePeriod time.Duration `validate:"min=1"`
-	Containers             []Container   `validate:"min=1"`
+	Version                string        `validate:"required,semantic_version"`
+	TerminationGracePeriod time.Duration `validate:"gt=0"`
+	Containers             []Container   `validate:"required,dive"`
```

Comment:
@lyracampos whats does `dive` means here?

Reply:
@GuilhermBrSp  **dive** means that the slice validations will be checked automatically.
in this case, all container slice validations will be checked whenever the scheduler is being validated

if you don't use dive in slice you'll need to call validations one by one for each slice item